### ISSUE TITLE
Can't use lookup bar to create a file with the same name of one I've previously deleted

### ISSUE THREAD

**Issue by ryan-p-randall** [dendronhq/dendron#592]
## Please select the area your Bug applys to. (Multiple selections are Possible)

- [ ] UI
- [ ] Workbench
- [ ] Publishing
- [ ] Pods
- [ ] Markdown
- [ ] Hierarchy
- [ ] Graph
- [x] Misc (Choose this if your not sure)

## Describe the bug

I tried to use the lookup bar to create a file using with the same name as one I've previously deleted, and Dendron gave me an an error instead of letting me create the file.

I noticed that a few files—specifically ones like `talk.template.notes.md` were showing up in the graph as isolated dots without any lines connecting them to other files. This isolation happened even after I referenced the note via a wikilink, like [[talk.template.notes]]. So I deleted the file and was going to add it back… but now I can't add it back.

After today's update to version 0.35, I no longer see the isolated "dot" notes, so I can't attach a screenshot. But I did attach the `dendron.log` file.

## To Reproduce

1. Open lookup bar (command-shift-s)
2. Type name of previously deleted file (talk.template.notes)
3. See error (something like "Dendron cannot create file", I think)

## Expected behavior

I expected to be able to make that file.

## Desktop (please complete the following information)

- OS: Mac 10.14.6
- VSCode 1.54.3
- Dendron 0.35

## Additional context

I have been getting an occasional error message, at seemingly random times that I can't anticipate. I copied it sometime in the last couple days and have attached it here.
[dendron-other-error.txt](https://github.com/dendronhq/dendron/files/6224342/dendron-other-error.txt)

## Dendron Log file

Please attach the output of `>Dev:Dendron: Open Logs` here
[dendron.log](https://github.com/dendronhq/dendron/files/6224307/dendron.log)

**Comment by af4jm** [comment]
I've noticed that any tine a file operation happens outside of a Dendron command, I will get all kinds of bizarre behavior until I reload the index... I haven't seen this specific issue, but I have seen the opposite where I copy-paste-rename a file, create a wikilink to it, theb hover that link & touch Ctrl... that triggers Dendron to create the file, overwriting whatever contents are there (there is an open issue for that already)

**Comment by kpathakota** [comment]
Yup, I think @af4jm hit the nail on the head there. The index must be reloaded anytime the file system changes underneath Dendron without the programs knowledge. The solution here is to use the Dendron: Delete Node command. I think there's some longer-term optimizations we could do that would listen for notifications from the file system but that seems out of scope for this right now. @ryan-p-randall - closing for now - please reopen if you think there's more to do here.

**Comment by ryan-p-randall** [comment]
Sorry I didn't make it clearer, @kpathakota & @af4jm , but I encountered this error specifically after having deleted the note **by using** the `Dendron: Delete Node` command. The operations with the note/node were done within Dendron's own tools, which is why this error was so confusing.

Here were the initial steps:
1. Use `Dendron: Delete Node` command to delete a file called `podcast.template.notes`
2. Try to use Dendron lookup bar to create `podcast.template.notes`
3. Receive error message

The above also happened for another note, called `talk.template.notes`.

What I've done to work around the problem since:
1. Use the Dendron lookup bar to create a note with a different name, something like `podcast.template.zombie-notes`
2. Use the `Dendron: Rename Note` command to rename `podcast.template.zombie-notes` to `podcast.template.notes`
3. (😄 because it worked!)

I've "fixed" the issue for my own notes, for now. So it's fine to leave this issue closed.

But in case it happens again, for myself or another user, I wanted to document that the index didn't correctly update for two nodes, even though I deleted them using Dendron's delete command. Also, both of these were referenced as templates in schema files, in case that's relevant.
